include/dronecore.h

```diff
@@ -10,40 +10,136 @@ namespace dronecore {
 
 class DroneCoreImpl;
 
+/**
+ * @brief This is the main class of **%DroneCore MAVLink API Library** (for the Dronecode Platform).
+ 
+ * It is used to discover vehicles and manage active connections.
+ *
+ * An instance of this class must be created (first) in order to use the library.
+ * The instance must be destroyed after use in order to break connections and release all resources.
+ */
 class DroneCore
 {
 public:
+
+    /**
+     * @brief Constructor. 
+     */
     DroneCore();
+
+    /**
+     * @brief Destructor.
+     *
+     * Disconnects all connected vehicles and releases all resources.
+     */
     ~DroneCore();
 
+    /**
+     * @brief Result type returned when adding a connection.
+     *
+     * **Note**: DroneCore does not throw exceptions. Instead a result of this type will be 
+     * returned when you add a connection: add_udp_connection().
+     */
     enum class ConnectionResult {
-        SUCCESS = 0,
-        TIMEOUT,
-        SOCKET_ERROR,
-        BIND_ERROR,
-        CONNECTION_ERROR,
-        NOT_IMPLEMENTED,
-        DEVICE_NOT_CONNECTED,
-        DEVICE_BUSY,
-        COMMAND_DENIED,
-        DESTINATION_IP_UNKNOWN,
-        CONNECTIONS_EXHAUSTED
+        SUCCESS = 0, /**< @brief %Connection succeeded. */
+        TIMEOUT, /**< @brief %Connection timed out. */
+        SOCKET_ERROR, /**< @brief Socket error. */
+        BIND_ERROR, /**< @brief Bind error. */
```

Comment:
@julianoes WOuld be good to have a useful string for all of these - by which I mean something that helps people work out why they might have got the error. Just sayin' :-)